src/matrix/net/RequestScheduler.ts

```diff
@@ -113,9 +140,7 @@ export class RequestScheduler {
                         request.methodName
                     ].apply(this._hsApi, request.args);
                     // so the request can be aborted
-                    request.requestResult = requestResult;
-                    const response = await requestResult.response();
-                    request.resolve(response);
+                    request.setRequestResult(requestResult);
```

Comment:
the await was important here as it prevents tight-looping and allows to catch any errors.